packages/perseus/src/widget-ai-utils/interactive-graph/prompt-utils.test.ts

```diff
@@ -0,0 +1,597 @@
+                [3, 1],
+            ],
+        };
+
+        const resultJSON = getPromptJSON(renderProps, userInput);
+
+        expect(resultJSON).toEqual({
+            type: "interactive-graph",
+            options: {
+                graph: {
+                    type: "segment",
+                    numSegments: 1,
+                    startCoords: [
+                        [0, 0],
+                        [1, 1],
+                    ],
+                },
+                backgroundImageUrl: "https://example.com/image.png",
+                range: [
+                    [-10, 10],
+                    [-10, 10],
+                ],
+                labels: ["x", "y"],
+            },
+            userInput: {
+                coords: [
+                    [2, 3],
+                    [3, 1],
+                ],
+            },
+        });
+    });
+
+    it("should return JSON for a sinusoid graph", () => {
+        const renderProps: any = {
+            graph: {
+                type: "sinusoid",
+                startCoords: [
+                    [0, 0],
+                    [1, 1],
+                ],
+            },
+            backgroundImage: {url: "https://example.com/image.png"},
+            range: [
+                [-10, 10],
+                [-10, 10],
+            ],
+            labels: ["x", "y"],
+        };
+
+        const userInput: any = {
+            type: "sinusoid",
+            coords: [
+                [2, 0],
+                [1, 3],
+            ],
+        };
+
+        const resultJSON = getPromptJSON(renderProps, userInput);
+
+        expect(resultJSON).toEqual({
+            type: "interactive-graph",
+            options: {
+                graph: {
+                    type: "sinusoid",
+                    startCoords: [
+                        [0, 0],
+                        [1, 1],
+                    ],
+                },
+                backgroundImageUrl: "https://example.com/image.png",
+                range: [
+                    [-10, 10],
+                    [-10, 10],
+                ],
+                labels: ["x", "y"],
+            },
+            userInput: {
+                coords: [
+                    [2, 0],
+                    [1, 3],
+                ],
+            },
+        });
+    });
+
+    it("should return JSON for an unsupported graph", () => {
+        const renderProps: any = {
+            graph: {
+                type: "fake-graph-type",
+            },
+            backgroundImage: {url: "https://example.com/image.png"},
+            range: [
+                [-10, 10],
+                [-10, 10],
+            ],
+            labels: ["x", "y"],
+        };
+
+        const userInput: any = {
+            type: "invalid",
+            coords: [
+                [0, 0],
+                [1, 1],
+            ],
+        };
+
+        const resultJSON = getPromptJSON(renderProps, userInput);
+
+        expect(resultJSON).toEqual({
```

Comment:
Stated elsewhere too, but I think if you use the `UnreachableCaseError` you can rely on the type system to catch new types of graphs instead of delaying the check till runtime. It also means you don't have to accommodate this type of "failure" at all.